main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/ministryofjustice/analytical-platform-auth0-user-exports/auth"
+	"log"
+	"os"
+	"time"
+
+	session2 "github.com/aws/aws-sdk-go/aws/session"
+	env "github.com/ministryofjustice/analytical-platform-auth0-user-exports/default_env_value"
+	connection "github.com/ministryofjustice/analytical-platform-auth0-user-exports/get_connection"
+	"github.com/ministryofjustice/analytical-platform-auth0-user-exports/job"
+)
+
+var (
+	apiUrl         = env.SetDefaultEnvValue("API_URL", "https://alpha-analytics-moj.eu.auth0.com")
+	clientId       = os.Getenv("CLIENT_ID")
```

Comment:
My usual advice when a configuration is strictly required is to blow up early with a meaningful error message rather then keep going and get a weird error message later on (as it's more confusing).

Specifically, what happens if `CLIENT_ID`/`CLIENT_SECRET` are not set and these values are empty strings?

I wonder if you could add another utility function to the `env` package to blowup when the variable is not there, e.g.:

```golang
function GetOrPanic(key, panicMsg string) {
  value, exists := := os.LookupEnv(key)
  if !exists {
    panic(panicMsg)
  }
  return value
}
```

Or even simpler/better:

```golang
function GetOrPanic(key) {
  value, exists := := os.LookupEnv(key)
  if !exists {
    panic(fmt.Sprintf(`"%s" environment variable not set but required.'`))
  }
  return value
}
```

so here you could say `clientSecret = env.GetOrPanic("CLIENT_SECRET")`.

Reply:
I handled it [here](https://github.com/ministryofjustice/analytical-platform-auth0-user-exports/blob/7010453e0f2eae33d1715838ae3be4cee91a45f2/auth/auth.go#L20)

Reply:
@xoen is this a good place or a ok way to do this?

Reply:
I'd be happy to have it raise here, I think would be clean but there are obviously infinite ways of doing anything.